interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/ExpressionCompiler.scala

```diff
@@ -200,30 +200,42 @@ class ExpressionCompiler(expressionParsers: Map[Language, ExpressionParser], dic
 
   private def substituteDictKeyExpression(expression: Expression, editor: Option[ParameterEditor], paramName: String)(
       implicit nodeId: NodeId
-  ) =
-    editor match {
-      case Some(DictParameterEditor(dictId)) =>
-        DictKeyWithLabelExpressionParser
-          .parseDictKeyWithLabelExpression(expression.expression)
-          .leftMap(errs => errs.map(_.toProcessCompilationError(nodeId.id, paramName)))
-          .andThen(expr =>
-            dictRegistry match {
-              case _: EngineDictRegistry =>
-                // no need to validate and resolve label it on Engine side, this allows EngineDictRegistry to be lighter (not having to contain dictionaries only used by DictParameterEditor)
-                Valid(expression)
-              case _ =>
-                dictRegistry
-                  .labelByKey(dictId, expr.key)
-                  .leftMap(e => NonEmptyList.of(e.toPartSubGraphCompilationError(nodeId.id, paramName)))
-                  .andThen {
-                    case Some(label) => Valid(Expression.dictKeyWithLabel(expr.key, Some(label)))
-                    case None => invalidNel(DictLabelByKeyResolutionFailed(dictId, expr.key, nodeId.id, paramName))
-                  }
-            }
-          )
-      case _ => Valid(expression)
+  ) = {
+    def substitute(dictId: String) = {
+      DictKeyWithLabelExpressionParser
+        .parseDictKeyWithLabelExpression(expression.expression)
+        .leftMap(errs => errs.map(_.toProcessCompilationError(nodeId.id, paramName)))
+        .andThen(expr =>
+          dictRegistry match {
+            case _: EngineDictRegistry =>
+              // no need to validate and resolve label it on Engine side, this allows EngineDictRegistry to be lighter (not having to contain dictionaries only used by DictParameterEditor)
+              Valid(expression)
+            case _ =>
+              dictRegistry
+                .labelByKey(dictId, expr.key)
+                .leftMap(e => NonEmptyList.of(e.toPartSubGraphCompilationError(nodeId.id, paramName)))
+                .andThen {
+                  case Some(label) => Valid(Expression.dictKeyWithLabel(expr.key, Some(label)))
+                  case None        => invalidNel(DictLabelByKeyResolutionFailed(dictId, expr.key, nodeId.id, paramName))
+                }
+          }
+        )
     }
 
+    if (expression.language == Language.DictKeyWithLabel && !expression.expression.isBlank && expression.expression != "null") // TODO: this isn't the prettiest check
```

Comment:
the "null" check can (and should be) removed, currently the issue is that FE instead of an empty string can send "null" if the field is cleared

Reply:
i've decided to remove it already, better to break it a little bit than add and remove ugly hotfixes